maat-court-data-api/src/test/java/gov/uk/courtdata/contributions/service/ContributionsServiceTest.java

```diff
@@ -0,0 +1,108 @@
+
+import gov.uk.courtdata.contributions.impl.ContributionsImpl;
+import gov.uk.courtdata.contributions.mapper.ContributionsMapper;
+import gov.uk.courtdata.dto.ContributionsDTO;
+import gov.uk.courtdata.entity.ContributionsEntity;
+import gov.uk.courtdata.exception.RequestedObjectNotFoundException;
+import gov.uk.courtdata.model.contributions.CreateContributions;
+import gov.uk.courtdata.model.contributions.UpdateContributions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.time.LocalDate;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatExceptionOfType;
+import static org.mockito.Mockito.*;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+@ExtendWith(MockitoExtension.class)
+class ContributionsServiceTest {
+
+    private static final Integer TEST_REP_ID = 999;
+
+    @InjectMocks
+    private ContributionsService contributionsService;
+    @Mock
+    private ContributionsImpl contributionsImpl;
+    @Mock
+    private ContributionsMapper contributionsMapper;
+
+    @Test
+    void whenFindIsInvoked_thenContributionsEntryIsRetrieved() {
+        when(contributionsImpl.findLatest(anyInt())).thenReturn(ContributionsEntity.builder().repId(TEST_REP_ID).build());
+        when(contributionsMapper.mapEntityToDTO(any(ContributionsEntity.class))).thenReturn(ContributionsDTO.builder().repId(TEST_REP_ID).build());
+
+        ContributionsDTO contributionsDTO = contributionsService.find(TEST_REP_ID);
+
+        assertThat(contributionsDTO.getRepId()).isEqualTo(TEST_REP_ID);
+    }
+
+    @Test
+    void givenContributionsEntryDoesntExist_whenFindIsInvoked_thenExceptionIsRaised() {
+        Integer testRepId = 666;
+        when(contributionsImpl.findLatest(anyInt())).thenReturn(null);
+
+        assertThatExceptionOfType(RequestedObjectNotFoundException.class)
+                .isThrownBy(() -> contributionsService.find(testRepId))
+                .withMessageContaining(String.format("Contributions entry not found for repId %d", testRepId));
+
+    }
+
+    @Test
+    void whenUpdateIsInvoked_thenContributionsEntryIsUpdated() {
+        Integer testId = 999;
+        ContributionsEntity contributionsEntity = ContributionsEntity.builder().build();
+        when(contributionsImpl.find(anyInt())).thenReturn(contributionsEntity);
+        when(contributionsImpl.update(any(ContributionsEntity.class))).thenReturn(contributionsEntity);
+        when(contributionsMapper.mapEntityToDTO(any(ContributionsEntity.class))).thenReturn(ContributionsDTO.builder().id(testId).build());
+
+        ContributionsDTO contributionsDTO = contributionsService.update(UpdateContributions.builder().id(testId).build());
+
+        assertThat(contributionsDTO.getId()).isEqualTo(testId);
+        verify(contributionsImpl).update(contributionsEntity);
+    }
```

Comment:
mock calls for `contributionsImpl.find()` and `contributionsMapper.mapEntityToDTO` can be verified. Same goes for all test cases where mock is used